ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/show/ShowMaterializedViewsFormatter.java

```diff
@@ -136,27 +135,19 @@ void showMaterializedViews(DataOutputStream out, List<Table> materializedViews)
 
   @NotNull
   private static String formatIncrementalRebuildMode(Table materializedView) {
-    String incrementalRebuild;
     HiveRelOptMaterialization relOptMaterialization = HiveMaterializedViewsRegistry.get().
-            getRewritingMaterializedView(materializedView.getDbName(), materializedView.getTableName(), ALL);
-    if (relOptMaterialization == null || relOptMaterialization.getRebuildMode() == UNKNOWN) {
-      incrementalRebuild = "Unknown";
-    } else {
-      switch (relOptMaterialization.getRebuildMode()) {
-        case AVAILABLE:
-          incrementalRebuild = "Available";
-          break;
-        case INSERT_ONLY:
-          incrementalRebuild = "Available in presence of insert operations only";
-          break;
-        case NOT_AVAILABLE:
-          incrementalRebuild = "Not available";
-          break;
-        default:
-          incrementalRebuild = "Unknown";
-          break;
-      }
+        getRewritingMaterializedView(materializedView.getDbName(), materializedView.getTableName(), ALL);
+    if (relOptMaterialization == null) {
+      return "Not available";
+    }
+    switch (relOptMaterialization.getRebuildMode()) {
+      case AVAILABLE:
+        return "Available";
```

Comment:
nit: does it make sense to extend the IncrementalRebuildMode enum with the return message for clarity?
like:
```
AVAILABLE ("Available"),
INSERT_ONLY("Available for insert operations only"),
...

```

Reply:
It is a good idea. I didn't do it earlier because I wanted to detach the formatter from the core logic but probably we'll use these messages in any other formatter and it can be changed anytime if it is needed.